rrmngmnt/filesystem.py

```diff
@@ -226,3 +227,193 @@ def wget(self, url, output_file, progress_handler=None):
+        :type fs: FileSystem
+        :param source: Full path to source
+        :type source: str
+        :param target: Path to target directory, if omitted, a temporary
+        folder is created instead
+        :type target: str
+        :param fs_type: File system type
+        :type fs_type: str
+        :param opts: Mount options separated by a comma such as:
+        'sync,rw,guest'
+        :type opts: str
+        """
+        super(MountPoint, self).__init__()
+        self.fs = fs
+        self.source = source
+        self.opts = opts
+        self.fs_type = fs_type
+        self.target = target
+        self._tmp = not bool(target)
+        self._mounted = False
+
+    def __enter__(self):
+        try:
+            self.mount()
+        except MountError as e:
+            self.logger.error(e)
+            raise
+        return self
+
+    def __exit__(self, type_, value, tb):
+        try:
+            self.umount()
+        except MountError as e:
+            self.logger.error(e)
+            raise
+
+    def __str__(self):
+        return (
+            """
+            Mounting point:
+            source: {source}
+            target: {target}
+            file system: {fs}
+            options: {opts}
+            """.format(
+                source=self.source,
+                target=self.target or "*tmp*",
+                fs=self.fs_type or "DEFAULT",
+                opts=self.opts or "DEFAULT",
+            )
+        )
+
+    def mount(self):
+        if self._tmp:
+            self.target = self.fs.mktemp(directory=True)
+        cmd = ['mount', '-v']
+        if self.fs_type:
+            cmd.extend(['-t', self.fs_type])
+        if self.opts:
+            cmd.extend(['-o', self.opts])
+        cmd.extend([self.source, self.target])
+        rc, out, err = self.fs.host.run_command(cmd)
+        if rc:
+            raise FailToMount(self, out, err)
+        self._mounted = True
+
+    def umount(self, force=True):
+        cmd = ['umount', '-v']
+        if force:
+            cmd.append('-f')
+        cmd.append(self.target)
+        rc, out, err = self.fs.host.run_command(cmd)
+        if rc:
+            raise FailToUmount(self, out, err)
+        if self._tmp and not self.fs.listdir(self.target):
+            self.fs.rmdir(self.target)
+        self._mounted = False
+
+    def remount(self, opts):
+        """
+        Remount disk
+
+        'remount' option is implicit
+        :param opts: Mount options separated by a comma such as:
+        'sync,rw,guest'
+        :type opts: str
+        """
+        if not self._mounted:
+            raise FailToRemount(self, '', 'not mounted!')
+        cmd = ['mount', '-v']
+        cmd.extend(['-o', 'remount,%s' % opts])
+        cmd.append(self.target)
+        rc, out, err = self.fs.host.run_command(cmd)
+        if rc:
+            raise FailToRemount(self, out, err)
+        self.opts = opts
+
+
+class FileSystemError(Exception):
```

Comment:
I would rather to see all exceptions in errors.py module. and have some hierarchy ... but I don't have anything specific in mind at this point ...